### PR TITLE
Add CI for image generation

### PR DESCRIPTION
**Pull request by Teebor-Choka**

### CODE REVIEW

File: .github/workflows/build-container.yaml [hunk 1]
```diff
@@ -8,53 +8,49 @@ on:
...
-      #     context: .
-      #     push: ${{ github.event_name != 'pull_request' }}
-      #     tags: evagalleryone/ai:latest
+        if: github.event_name != 'pull_request'
+        uses: bentoml/containerize-push-action@v1
+        with:
+          context: ./
+          bentofile: ./bentofile.yaml
+          bento-tag: evagallery_ai_api:latest
+          push: true
```
Comment (Teebor-Choka) [inline .github/workflows/build-container.yaml]:
```suggestion
        uses: bentoml/containerize-push-action@v1
        with:
          context: ./
          bentofile: ./bentofile.yaml
          bento-tag: evagallery_ai_api:latest
          push: {{ github.event_name != 'pull_request' }}
```

File: setup.py [hunk 2]
```diff
@@ -10,6 +10,13 @@
 logger = get_logger()
 
 
+if not settings.model.pre_download_during_build:
+    logger.info("Skipping model pre-download as it is disabled in settings.yaml")
+    exit(0)
+else:
+    logger.info("Pre-downloading models... WARNING: You likely don't want to do this!! This will make the image large.")
+
+
```
Comment (Teebor-Choka) [inline setup.py]: Please, nest the rest of the code under else and drop the exit(0).

Comment (Teebor-Choka) [reply setup.py]: Will the model be downloaded into the image partition in `/cache`? it should then be attached as external storage then.

Comment (slobodaapl) [reply setup.py]: This is a part of the build process, but the environment variable EVA_AI_MODEL__CACHE_DIR controls where the models are downloaded. Ideally, you should set the "download at build" env var to false in github vars. It's there just for testing.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -5,24 +5,65 @@ on:
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   test:
+    timeout-minutes: 15
     runs-on: ubuntu-latest
+    env:
+      CI: true
```
Comment (Teebor-Choka) [inline .github/workflows/test.yaml]: This is present by default.

**Comment by Teebor-Choka** [comment]
Why is the build workflow not triggered? Concurrency with the same identifier?

File: setup.py (hunk 2 above)
Comment (Teebor-Choka) [reply setup.py]: Sure, I'm asking because of the eventual deployment.

**Comment by slobodaapl** [comment]
> Why is the build workflow not triggered? Concurrency with the same identifier?

Working on it, not sure

**Pull request merged**